### ISSUE TITLE
bindings[:view] is nil in enum block

### ISSUE THREAD

**Issue by jandillmann** [sferik/rails_admin#2684]
According to the [documentation on Enumerations](https://github.com/sferik/rails_admin/wiki/Enumeration) it should be possible to do this:

    field :post_id, :enum do
      enum do
        Posts.accessible_by(bindings[:view]._current_user).map { |p| [p.title, p.id] }
      end
    end

I'm using this (virtual) enum field as a filter for a belongs_to association with a dropdown, but each user should only see their own posts in the dropdown:

    list do
      filters [:post_id]
      field :post do
        filterable false
      end
      field :title
    end

However, `bindings[:view]` is nil inside the `enum do` block…

**Comment by thedanielhanke** [comment]
just a comment: when using cancan, which i assume you do (`.assessible_by`), the ability-class should allow you to get exactly what you want by using the actual association.

**Comment by jandillmann** [comment]
@thedanielhanke How would that look like? Doesn't the ability class also need the current user?

**Comment by thedanielhanke** [comment]
sorry for the delay @jandillmann  .

well, my idea was, that you  specify like this:
`can :read, Post, user_id: user.id`

in your ability class. however, using the controller works fine for me:

```
field :post_id, :enum do
  enum do
    Posts.accessible_by(bindings[:controller].current_user).map { |p| [p.title, p.id] }
  end
end
```

btw: may i ask why you use a enum here and not a belongs_to association?

**Comment by jandillmann** [comment]
I use the enum field as a filter for a belongs_to relationship (see https://github.com/sferik/rails_admin/issues/2684#issue-169613759), so I can select the objects from a dropdown – otherwise the filter only shows a text box.

Unfortunately, in this context `bindings[:controller]` is also nil, so I can't get the current user and therefore can't filter the posts by accessibility.

**Comment by thedanielhanke** [comment]
@jandillmann strange. is there a way to get/see more of your model, maybe even sample-data?
if you`d commit your project (relevant parts) i can offer to check it out and try to assist.

cheers

**Comment by jandillmann** [comment]
Thanks for your help @thedanielhanke !

I uploaded a small demo project: https://github.com/jandillmann/rails-admin-demo

The relevant line is: https://github.com/jandillmann/rails-admin-demo/blob/master/app/models/concerns/post_admin.rb#L11

**Comment by thedanielhanke** [comment]
@jandillmann what i just noticed and overread until now:
`user_id` or `post_id` both dont point at the association. thats why you get a textfield. 
using `field :user` or `field :post` instead gives you that desired dropdown.

and question: since using your code untouched, the bindings are not nil in my case. so i wonder: are you using better-errors/binding of caller on the actual project to check the values of bindings? if so, bindings is a in-use name. 
```
enum do
  controller = bindings[:controller]
  raise
```
and controller should have a value in better-errors, `bindings` is still `nil`.

the closest to your code in the demo, working as (i understood):
```ruby
    rails_admin do
      object_label_method :title

      field :user do
        visible false
        default_value do
          bindings[:controller].current_user.id
        end
      end

      list do
        filters [:user]
        field :id
        field :user
        field :title
      end
```


the rest is handled by cancancan.

can you use that?

**Comment by jandillmann** [comment]
Now I'm confused… Both your suggestions don't work for me, neither with rails_admin version 1.0.0 nor with 1.1.0.

When setting `controller = bindings[:controller]` the variable is `nil`, in better-errors as well as when trying to use `controller.current_user` inside the enum block.

And using `:user` instead of `:user_id` gives me a textbox instead of a dropdown.

Could this be a Ruby or Rails version issue? I'm running Ruby 2.3.1 and JRuby 9.1.5.0 here with Rails 4.2.7.

**Comment by thedanielhanke** [comment]
well i run it against your gemfile using 2.3.1.
did you try to restart your rails app after changing the code? sometimes RA does not properly reload the block-contents

**Comment by jandillmann** [comment]
Yes, I already noticed that, so the `rails server` command is exited and restarted every time.
I've updated the demo project with more information, maybe this makes it more clear what I'm trying to achieve: https://github.com/jandillmann/rails-admin-demo/blob/master/app/models/concerns/post_admin.rb

**Comment by thedanielhanke** [comment]
try
```ruby
edit do
        field :title
        field :body
        field :user_id, :hidden do

          # This doesn't seem to do anything, the filter text field is empty by default
          default_value do
            bindings[:controller].current_user.id
          end
        end
      end
```

**Comment by jandillmann** [comment]
Sorry, I think we got sidetracked. This is not the part in question, I just copied and pasted the whole code section from your answer to the demo project. When creating new posts the `user_id` is automatically set to the current user, this part works just fine.

What doesn't work is the user filter on the post's index action where I'd like a dropdown of all users accessible by the current user, not a text input field.

**Comment by ricardoccpaiva** [comment]
guys,, can you help me figure out something ?
is it possible to have a sort of "hard coded" filter for a specific field ? i don't want user input on that , just want ti have de ability to filter column X by whatever hardcoded value i want.
Thanks

**Comment by ricardoccpaiva** [comment]
for instance.. i want to filter a list of documents by user id (obtained by the user who's logged in)

**Comment by rushil1999** [comment]
What is bindings[:view]?
